processor/swok8sworkloadstatusprocessor/processor.go

```diff
@@ -0,0 +1,239 @@
+// Copyright 2025 SolarWinds Worldwide, LLC. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package swok8sworkloadstatusprocessor // import "github.com/solarwinds/solarwinds-otel-collector-contrib/processor/swok8sworkloadstatusprocessor"
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"strings"
+
+	"go.opentelemetry.io/collector/component"
+	"go.opentelemetry.io/collector/pdata/pcommon"
+	"go.opentelemetry.io/collector/pdata/plog"
+	"go.uber.org/zap"
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/client-go/informers"
+	"k8s.io/client-go/tools/cache"
+
+	"github.com/solarwinds/solarwinds-otel-collector-contrib/processor/swok8sworkloadstatusprocessor/internal"
+)
+
+type swok8sworkloadstatusprocessor struct {
+	logger      *zap.Logger
+	podInformer cache.SharedIndexInformer
+	cancel      context.CancelFunc
+	config      *Config
+	factory     informers.SharedInformerFactory
+}
+
+func (p *swok8sworkloadstatusprocessor) Start(_ context.Context, _ component.Host) error {
+
+	p.logger.Info("Starting swok8sworkloadstatus processor")
+	cancelCtx, cancelFunc := context.WithCancel(context.Background())
+	p.cancel = cancelFunc
+
+	client, err := p.config.getK8sClient()
+	if err != nil {
+		return err
+	}
+
+	p.factory = informers.NewSharedInformerFactory(client, p.config.WatchSyncPeriod)
+
+	p.podInformer = p.factory.Core().V1().Pods().Informer()
```

Comment:
Later, we might want to create some kind of an extension that would allow us to share this cache across processors and their instances. We are creating way too many subscriptions to the k8s API server.